YAFC/Workspace/ProductionTable/ProductionTableFlatHierarchy.cs

```diff
@@ -164,19 +194,36 @@ public class FlatHierarchy<TRow, TGroup> where TRow : ModelObject<TGroup>, IGrou
         private void Rebuild() {
             flatRecipes.Clear();
             flatGroups.Clear();
+            rowHighlighting.Clear();
+
             BuildFlatHierarchy(root);
             rebuildRequired = false;
         }
 
-        private void BuildFlatHierarchy(TGroup table) {
+        private void BuildFlatHierarchy(TGroup table, RowHighlighting parentHighlight = RowHighlighting.None) {
             foreach (var recipe in table.elements) {
                 flatRecipes.Add(recipe);
+
+                RowHighlighting highlight = parentHighlight;
+
+                if (recipe is RecipeRow r && r.highlighting != RowHighlighting.None) {
+                    // Only respect any highlighting if the recipe is in fact a RecipeRow
+                    highlight = r.highlighting;
+                }
+
+                rowHighlighting.Add(highlight);
+
                 var sub = recipe.subgroup;
-                if (sub != null && sub.expanded) {
+
+                if (sub is { expanded: true }) {
                     flatGroups.Add(sub);
-                    BuildFlatHierarchy(sub);
+
+                    BuildFlatHierarchy(sub, highlight); // Pass the current highlight color to the child rows
+
                     flatRecipes.Add(null);
                     flatGroups.Add(sub);
+
+                    rowHighlighting.Add(RowHighlighting.None);
```

Comment:
Why do you need to add another 'row highlighting'? Is there a 'hidden row' between a nested recipe and the parent one or something? (I see that `flatRecipes` and `flagGroups` (directly above) do something similar)

It might be good to add a comment explaining this, as it is not clear from the code.

Reply:
Yes, there seems to be a hidden row. For separation purposes, I guess. But I'm not sure how to explain this and if I even got it right.

I'll investigate this and add a comment that explains this.

Reply:
Done. See #91